src/Fantomas/CodePrinter.fs

```diff
@@ -1029,10 +1029,29 @@ and genExpr astContext synExpr =
             +> indent +> (ifElse (not hasPar && addSpaceBefore) sepSpace sepNone) +> appNlnFun e2 (genExpr astContext e2) +> unindent)
 
     // Always spacing in multiple arguments
-    | App(e, es) -> 
-        atCurrentColumn (genExpr astContext e +> 
+    | App(e, es) ->
+        // https://github.com/fsprojects/fantomas/issues/545
+        // we need to make sure each expression in the function application has offset at least greater than
+        // identation of the function expression itself
+        // we replace sepSpace in such case
+        let mutable savedColumn = 0
+        let retainColumn ctx =
+            savedColumn <- ctx.Writer.Column
+            ctx
+        let indentIfNeeded ctx =
+            let savedColumn = savedColumn
```

Comment:
```suggestion
            let savedColumn = ctx.Writer.AtColumn
```

`AtColumn` can be used here to same effect, removing need for mutable variable.

Reply:
Thanks, I’ll give it a go!

Reply:
It works as you mentionned, thanks for the review.